### PR TITLE
Maintenance/cleanup api server

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

During the design of this first version, there were many variants of the design that remained in the code. This PR cleans up all the deprecated code before extending it with new features. Some relevant changes:

- Pruned and upgraded requirements
- Removed trials ``sandbox`` folder
- Removed unused models, db repos and routes
- Left only basic authentication
- Added client wrapper for webserver sessions (see osparc-simcore/services/api-server/src/simcore_service_api_server/services/webserver.py)
- Reverted logger from loguru to default python logger
- documented dev: ``services/api-server/README.md``

Some of the code that was removed will probably be reintegrated when actually needed (e.g. other authentication schemes)

## How to test

unit testing in
```cmd
make devenv
cd services/api-tests
make install-dev tests
```
and read ``README.md`` for manual tests of the API

## Checklist

- [x] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [x] Unit tests for the changes exist (partially)
- [ ] Runs in the swarm
- [x] Documentation reflects the changes
- [x] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

File: services/api-server/src/simcore_service_api_server/api/dependencies/authentication.py [hunk 1]
```diff
@@ -1,3 +1,48 @@
...
+from ...db.repositories.users import UsersRepository
+from .database import get_repository
+
+# SEE https://swagger.io/docs/specification/authentication/basic-authentication/
+basic_scheme = HTTPBasic()
+
+
+def _create_exception():
+    _unauthorized_headers = {
+        "WWW-Authenticate": f'Basic realm="{basic_scheme.realm}"'
```
Comment (sanderegg) [inline services/api-server/src/simcore_service_api_server/api/dependencies/authentication.py:14]: are you sure about how to capitalize this string? I saw many different ones in the different APIs some times with Www-authenticate, sometimes all lower case, ...

File: services/api-server/tests/unit/test_settings.py [hunk 2]
```diff
@@ -14,7 +14,9 @@
 # bring .env-devel in here
 def test_min_environ_for_settings(monkeypatch):
     monkeypatch.setenv("WEBSERVER_HOST", "production_webserver")
-    monkeypatch.setenv("WEBSERVER_SESSION_SECRET_KEY", "REPLACE ME with a key of at least length 32.")
+    monkeypatch.setenv(
+        "WEBSERVER_SESSION_SECRET_KEY", "REPLACE ME with a key of at least length 32."
+    )
```
Comment (sanderegg) [inline services/api-server/tests/unit/test_settings.py:19]: one stupid question, what is the difference between monkeypatching and using the mock library? as far as I understand this monkeypatch fixture you can also replace functions right?

File: services/api-server/src/simcore_service_api_server/api/dependencies/authentication.py (hunk 1 above)
Comment (pcrespov) [reply services/api-server/src/simcore_service_api_server/api/dependencies/authentication.py:14]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate

File: services/api-server/tests/unit/test_settings.py (hunk 2 above)
Comment (pcrespov) [reply services/api-server/tests/unit/test_settings.py:19]: they way i see it is that monkeypatch is more high-level in the sense that e.g. you can replace easily environment variables in a simple call while mock is more lower level

See examples here https://docs.pytest.org/en/latest/monkeypatch.html

**Pull request merged**